src/gameplay/Game.js

```diff
@@ -111,6 +111,7 @@ class Game {
 
     if (!spell.canInvoke(this, player, effectParams)) return Game.error('cannot invoke');
     spell.invoke(this, player, effectParams);
+    player.hasActivated[spell.Name] = true;
```

Comment:
`spell.Name` cannot get the card's name (`XxxxCard.Name` is a static field).
You should use spell.name